spec/classes/datadog_agent_integrations_mysql_spec.rb

```diff
@@ -43,71 +43,71 @@
+      context 'with parameters set' do
+        let(:params) do
+          {
+            password: 'foobar',
+            host: 'mysql1',
+            user: 'baz',
+            sock: '/tmp/mysql.foo.sock',
+            replication: '1',
+            galera_cluster: '1',
+          }
         end
 
-        context 'with tags parameter array' do
-          let(:params) do
-            {
-              password: 'foobar',
-              tags: ['foo', 'bar', 'baz'],
-            }
-          end
+        it { is_expected.to contain_file(conf_file).with_content(%r{pass: foobar}) }
+        it { is_expected.to contain_file(conf_file).with_content(%r{server: mysql1}) }
+        it { is_expected.to contain_file(conf_file).with_content(%r{user: baz}) }
+        it { is_expected.to contain_file(conf_file).with_content(%r{sock: /tmp/mysql.foo.sock}) }
+        it { is_expected.to contain_file(conf_file).with_content(%r{replication: 1}) }
+        it { is_expected.to contain_file(conf_file).with_content(%r{galera_cluster: 1}) }
+      end
 
-          it { is_expected.to contain_file(conf_file).with_content(%r{tags:[^-]+- foo\s+- bar\s+- baz\s*?[^-]}m) }
+      context 'with tags parameter array' do
+        let(:params) do
+          {
+            password: 'foobar',
+            tags: ['foo', 'bar', 'baz'],
+          }
         end
 
-        context 'tags not array' do
-          let(:params) do
-            {
-              password: 'foobar',
-              tags: 'aoeu',
-            }
-          end
+        it { is_expected.to contain_file(conf_file).with_content(%r{tags:[^-]+- foo\s+- bar\s+- baz\s*?[^-]}m) }
+      end
 
-          it { is_expected.not_to compile }
+      context 'tags not array' do
+        let(:params) do
+          {
+            password: 'foobar',
+            tags: 'aoeu',
+          }
         end
-      end
-    end
 
-    context 'with queries parameter set' do
-      let(:params) do
-        {
-          password: 'foobar',
-          queries: [
-            {
-              'query'  => 'SELECT TIMESTAMPDIFF(second,MAX(create_time),NOW()) as last_accessed FROM requests',
-              'metric' => 'app.seconds_since_last_request',
-              'type'   => 'gauge',
-              'field'  => 'last_accessed',
-            },
-          ],
-        }
+        it { is_expected.not_to compile }
       end
 
-      it { is_expected.to contain_file(conf_file).with_content(%r{- query}) }
-      it { is_expected.to contain_file(conf_file).with_content(%r{query: SELECT TIMESTAMPDIFF\(second,MAX\(create_time\),NOW\(\)\) as last_accessed FROM requests}) }
-      it { is_expected.to contain_file(conf_file).with_content(%r{metric: app.seconds_since_last_request}) }
-      it { is_expected.to contain_file(conf_file).with_content(%r{type: gauge}) }
-      it { is_expected.to contain_file(conf_file).with_content(%r{field: last_accessed}) }
+      context 'with queries parameter set' do
+        let(:params) do
```

Comment:
Why do we want `let` here (and elsewhere in this file)?

Reply:
That's the way we set the arguments for the integrations. If there's a better way to do it, I don't know it 😅